File: SocialAppFlutter/social_app/lib/Models/attended_location.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:social_app/Models/location.dart';
import 'package:social_app/Models/sport_category.dart';

part 'json_generated_code/attended_location.g.dart';

@JsonSerializable(explicitToJson: true)
class AttendedLocation {
  final Location location;
  final SportCategory sportCategory;
  final DateTime attendedDatetime;

  AttendedLocation(this.location, this.sportCategory, this.attendedDatetime);

  factory AttendedLocation.fromJson(Map<String, dynamic> json) =>
      _$AttendedLocationFromJson(json);

  Map<String, dynamic> toJson() => _$AttendedLocationToJson(this);
}

File: APIs/EventsService/EventsService/DTOs/AttendedLocation.cs
using Newtonsoft.Json;

namespace EventsService.DTOs
{
    public class AttendedLocation
    {
        [JsonProperty("location")]
        public LocationObject Location { get; set; } = null!;
        [JsonProperty("sportCategory")]
        public SportCategoryObject SportCategory { get; set; } = null!;
        public DateTime AttendedDatetime { get; set; }

    }
}
